nucypher/blockchain/eth/actors.py

```diff
@@ -318,9 +302,10 @@ def connect_condition_providers(
                 duplicated_endpoint_check[chain_id].add(uri)
 
         # Ingest default/fallback RPC providers for each chain
-        for chain_id in self.domain.condition_chain_ids:
-            default_endpoints = get_healthy_default_rpc_endpoints(chain_id)
-            for uri in default_endpoints:
+        # for chain_id in self.domain.condition_chain_ids:
+        default_endpoints = get_healthy_default_rpc_endpoints(self.domain.name)
+        for chain_id in default_endpoints:
+            for uri in default_endpoints[chain_id]:
```

Comment:
Taking a second look at the code, let's do the random shuffle here instead of in L173 in eth/utils.py:
1. makes things cleaner by keeping separate logic in that utils.py can just be focused on getting the endpoints, and not on the specifics of using them
2. we only shuffle healthy endpoints
3. more efficient since it saves us the extra iteration through the dictionary done on eth/utils.py L172 - we are iterating here anyway.
4. the shuffle logic is the same irrespective of domain

```suggestion
            endpoints = default_endpoints[chain_id]
            # randomize list so that the same fallback RPC endpoints aren't always used by all nodes
            random.shuffle(endpoints)
            for uri in endpoints:
```

Reply:
done